Source/FieldSolver/WarpXPushFieldsHybridPIC.cpp

```diff
@@ -238,11 +238,52 @@ void WarpX::HybridPICDepositRhoAndJ ()
 
     // Perform charge deposition in component 0 of rho_fp at current time.
     mypc->DepositCharge(m_fields.get_mr_levels(FieldType::rho_fp, finest_level), 0._rt);
+
     // Perform current deposition at t_{n-1/2}.
-    mypc->DepositCurrent(m_fields.get_mr_levels_alldirs(FieldType::current_fp, finest_level), dt[0], -0.5_rt * dt[0]);
+    // This is broken up into deposition across multiple species MultiFabs to preserve
+    // Species average velocities, whic is used in collision processes.
+
+    auto & J_all = m_fields.get_mr_levels_alldirs(FieldType::current_fp, finest_level);
+
+    // Zero all components of the total current density array.
+    for (const auto& J_lev : J_all)
+    {
+        J_lev[0]->setVal(0.0_rt);
+        J_lev[1]->setVal(0.0_rt);
+        J_lev[2]->setVal(0.0_rt);
+    }
+
+    for (const auto & spec : mypc->GetSpeciesNames())
+    {
+        auto & J_species = m_fields.get_mr_levels_alldirs("hybrid_current_fp_" + spec, finest_level);
+
+        // Retrieve species specific array for deposition
+        mypc->DepositSpeciesCurrent(J_species, dt[0], -0.5_rt * dt[0]);
+
+        // Filter and sychronize current for each species
+        SyncCurrent("hybrid_current_fp_" + spec);
+
+        for (int lev = 0; lev <= finest_level; ++lev)
+        {
+            ApplyJfieldBoundary(lev,
+                m_fields.get("hybrid_current_fp_" + spec, Direction{0}, lev),
```

Comment:
I think we can drop the `hybrid` at the start of the name, and then this overall functionality would be easy to extend to generally allow diagnostic output of the individual species currents, which is something I've been meaning to do for a while.

Reply:
ok, that works for me. I can adjust the MF naming convention accordingly.